core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1560,26 +1567,30 @@ private[spark] class BlockManager(
   def replicateBlock(
       blockId: BlockId,
       existingReplicas: Set[BlockManagerId],
-      maxReplicas: Int): Unit = {
+      maxReplicas: Int,
+      forceFetch: Boolean = true,
```

Comment:
Why are we making this configurable? And also can we improve the parameter name since this is just controlling force fetching of the peer list.

Reply:
Parameter name changes to forceFetchPeers.

To decommision this block managers, we have to replicate all the cache blocks to some other peers. To replicate n blocks, we are going to call this method n times. This method force fetches the peers each time it is called. So it will end up making n getPeer calls which are not required. So we have made it configurable (by default to true). The decommission flow refreshes the peer list once in beginning and then calls replicateBlock(forceFetchPeers=false) to avoid n calls.

Reply:
forceFetch is not needed to be configurable anymore.